src/logger/storage.rs

```diff
@@ -23,11 +23,19 @@ pub struct Storage<'a> {
     pub values: HashMap<&'a str, serde_json::Value>,
 }
 
-impl Storage<'_> {
+impl<'a> Storage<'a> {
     /// Default constructor.
     pub fn new() -> Self {
         Self::default()
     }
+
+    pub fn record_value(&mut self, key: &'a str, value: serde_json::Value) {
+        if super::formatter::IMPLICIT_KEYS.contains(key) {
+            tracing::warn!("{key} is a reserved entry. Skipping it. value: {value}");
```

Comment:
is this safe to call here?

Reply:
warn? yes, tested it with duplicate keys, this just logs the warn message and won't cause stack overflow

Reply:
sure